src/main/java/com/hatstone/bobertapi/controller/SubmissionController.java

```diff
@@ -24,12 +28,15 @@ public class SubmissionController {
     @PostMapping("/create-submission")
     public ResponseEntity<Long> CreateSubmission(@RequestBody Submission submission){
         String insertQuery = "INSERT INTO submissions (userid, problemid, data, language) VALUES(?,?,?,?)";
+        String fetchArgsQuery = "SELECT inputargs FROM testcases INNER JOIN submissions ON testcases.problemid = ?";
```

Comment:
I don't actually think we need an inner join here, which I didn't catch last time due to having a brain so foggy it might as well have been the entrance to a haunted house. a simpler query would be `SELECT inputargs FROM testcases WHERE problemid = ?`, and then later we can just replace that question mark with `submission.getProblemId()`, which we grab externally and therefore don't need a join for.

Reply:
fixed